spyder/plugins/projects/plugin.py

```diff
@@ -148,251 +100,169 @@ def on_initialize(self):
         """Register plugin in Spyder's main window"""
         widget = self.get_widget()
         treewidget = widget.treewidget
+        self._completions = None
 
-        self.ipyconsole = None
-        self.editor = None
-        self.completions = None
+        # Emit public signals so that other plugins can connect to them
+        widget.sig_project_created.connect(self.sig_project_created)
+        widget.sig_project_closed.connect(self.sig_project_closed)
+        widget.sig_project_loaded.connect(self.sig_project_loaded)
 
         treewidget.sig_delete_project.connect(self.delete_project)
         treewidget.sig_redirect_stdio_requested.connect(
             self.sig_redirect_stdio_requested)
         self.sig_switch_to_plugin_requested.connect(
-            lambda plugin, check: self.show_explorer())
-        self.sig_project_loaded.connect(self.update_explorer)
+            lambda plugin, check: self._show_main_widget())
 
         if self.main:
             widget.sig_open_file_requested.connect(self.main.open_file)
-            self.sig_project_loaded.connect(
+            widget.sig_project_loaded.connect(
                 lambda v: self.main.set_window_title())
-            self.sig_project_closed.connect(
+            widget.sig_project_closed.connect(
                 lambda v: self.main.set_window_title())
             self.main.restore_scrollbar_position.connect(
-                self.restore_scrollbar_position)
+                self.get_widget().restore_scrollbar_position)
 
         self.register_project_type(self, EmptyProject)
-        self.setup()
 
     @on_plugin_available(plugin=Plugins.Editor)
     def on_editor_available(self):
-        self.editor = self.get_plugin(Plugins.Editor)
+        editor = self.get_plugin(Plugins.Editor)
         widget = self.get_widget()
         treewidget = widget.treewidget
 
-        treewidget.sig_open_file_requested.connect(self.editor.load)
-        treewidget.sig_removed.connect(self.editor.removed)
-        treewidget.sig_tree_removed.connect(self.editor.removed_tree)
-        treewidget.sig_renamed.connect(self.editor.renamed)
-        treewidget.sig_tree_renamed.connect(self.editor.renamed_tree)
-        treewidget.sig_module_created.connect(self.editor.new)
-        treewidget.sig_file_created.connect(self._new_editor)
+        editor.set_projects(self)
 
-        self.sig_project_loaded.connect(self._setup_editor_files)
-        self.sig_project_closed[bool].connect(self._setup_editor_files)
+        treewidget.sig_open_file_requested.connect(editor.load)
+        treewidget.sig_removed.connect(editor.removed)
+        treewidget.sig_tree_removed.connect(editor.removed_tree)
+        treewidget.sig_renamed.connect(editor.renamed)
+        treewidget.sig_tree_renamed.connect(editor.renamed_tree)
+        treewidget.sig_module_created.connect(editor.new)
+        treewidget.sig_file_created.connect(self._new_editor)
```

Comment:
Maybe the projects `main_widget` could expose these signals? So from the plugin is not necessary to know that a `treewidget` like widget is available? So for example something like:

```python
widget.sig_file_created.connect(self._new_editor)
```

If worthy to change, then something similar will be needed in other places where signals for the `treewidget` widget connection is done.